BitwardenShared/UI/Tools/Generator/Generator/GeneratorProcessor.swift

```diff
@@ -348,9 +348,7 @@ final class GeneratorProcessor: StateProcessor<GeneratorState, GeneratorAction,
         var passwordOptions = state.passwordState.passwordGenerationOptions
         state.isPolicyInEffect = await (try? services.policyService
             .applyPasswordGenerationPolicy(options: &passwordOptions)) ?? false
-        // When updating the state, don't update the password generator type or it can override the
-        // user's current selection.
-        state.passwordState.update(with: passwordOptions, shouldUpdateGeneratorType: false)
+        state.passwordState.update(with: passwordOptions, shouldUpdateGeneratorType: true)
```

Comment:
🤔 Can't the scenario in the comment happen anymore? I mean isn't this line with `true` value already called in `loadGeneratorOptions()`? Is it necessary to also have it here?

Reply:
This new behaviour should enforce the user to use the `generatorType` from policies, so the scenario we were trying to avoid, now it is mandatory.

This was causing sync to not update the `generatorType`

Reply:
Great makes sense thanks!